src/audio/selector/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
+if(CONFIG_IPC_MAJOR_3)
 add_local_sources(sof selector_generic.c selector.c)
+else()
+add_local_sources(sof ../module_adapter/module_adapter.c ../module_adapter/module/generic.c selector_generic.c selector.c)
```

Comment:
Typically people search for information what sources are compiled in the directory where whey are placed (typically CMakeLists.txt in that directory contains information about sources used).
Please move part of this else clause to `src/audio/module_adapter/CMakeLists.txt` under
```
if (NOT CONFIG_IPC_MAJOR_3)
    add_local_sources(sof module_adapter.c module/generic.c)
endif()
```